content/en/blog/2022-11-11-prove-the-authenticity-of-helm-charts/index.md

```diff
@@ -0,0 +1,281 @@
+{{% /note %}}
+
+As we stick to GitOps practices, we only create files that contain the `HelmRepository` and `HelmRelease` resources. After committing and pushing those changes into the upstream repository, Flux will watch for changes and use them as source-of-truth for the configuration:
+
+```shell
+git clone git@github.com:developer-guy/flux-cosign-helm-oci-demo.git
+cd flux-cosign-helm-oci-demo
+```
+
+Let's create the _HelmRepository_ resource first:
+
+```shell
+flux create source helm prometheus-community \
+    --url=oci://ghcr.io/prometheus-community/charts \
+    --interval=10m \
+    --export > ./clusters/my-cluster/prometheus-community-helmrepository.yaml
+```
+
+Now, let's move on with creating the _HelmRelease_ resource:
+
+```shell
+flux create helmrelease prometheus \
+    --source=HelmRepository/prometheus-community \
+    --chart=prometheus \
+    --interval=10m \
+    --release-name prometheus \
+    --target-namespace=monitoring \
+    --create-target-namespace \
+    --export > ./clusters/my-cluster/prometheus-helmrelease.yaml
+```
+
+and run the following command to add the `verify` section to the _HelmRelease_ resource' `.spec.chart.spec` section  to enable keylesss verification:
+
+```shell
+yq e '.spec.chart.spec|=({"verify": { "provider": "cosign" } } +.)' ./clusters/my-cluster/prometheus-helmrelease.yaml
+```
+
+then, commit and push the changes:
+
+```shell
+git commit -m "Add prometheus HelmRelease and HelmRepository resources"
+git push
+```
+
+After a couple of seconds, Flux will have applied these changes. Now let's check the status of them:
+
+> Or, you can trigger the reconciliation immediately by running the simple command: `flux reconcile source git flux-system`
+
+For the _HelmRepository_ resource:
+
+```shell
+$ flux get sources helm
+NAME                    REVISION        SUSPENDED       READY   MESSAGE
+prometheus-community                    False           True    Helm repository is ready
+```
+
+For the _HelmRelease_ resource:
+
+```shell
+$ flux get helmreleases
+NAME            REVISION        SUSPENDED       READY   MESSAGE
+prometheus      15.18.0         False           True    Release reconciliation succeeded
+```
+
+If everything is fine, you can check the pods in the `monitoring` namespace:
+
+```shell
+$ kubectl get pods -n monitoring
+NAME                                             READY   STATUS    RESTARTS   AGE
+prometheus-alertmanager-54b7d7cf45-2b7zf         2/2     Running   0          115s
+prometheus-kube-state-metrics-67f68d64bb-vlmvd   1/1     Running   0          115s
+prometheus-node-exporter-46gm6                   1/1     Running   0          115s
+prometheus-pushgateway-596cd99697-t79zt          1/1     Running   0          115s
+prometheus-server-c458cf6f9-nvstw                2/2     Running   0          115s
+```
+
+Great! Now, you have installed Prometheus with Flux by using the Helm chart stored in the OCI registry and verified it with _cosign_.
```

Comment:
Something feels missing here.
How can the user tell if something is verified or not?
Maybe showing the HelmChart status where we show that a given chart is verified may help.
Or maybe just mention what would happen in the verification fails, HelmRelease failure.